atd-etl/cris_import/cris_import.py

```diff
@@ -0,0 +1,821 @@
+        )
+        ssh_tunnel.start()
+
+        pg = psycopg2.connect(
+            host="localhost",
+            port=ssh_tunnel.local_bind_port,
+            user=DB_USER,
+            password=DB_PASS,
+            dbname=DB_NAME,
+            sslmode=DB_SSL_REQUIREMENT,
+            sslrootcert="/root/rds-combined-ca-bundle.pem",
+        )
+
+        cursor = pg.cursor()
+
+        # check if the schema exists by querying the pg_namespace system catalog
+        cursor.execute(
+            f"SELECT EXISTS (SELECT 1 FROM pg_namespace WHERE nspname = '{map_state['import_schema']}')"
+        )
+
+        schema_exists = cursor.fetchone()[0]
+
+        # if the schema doesn't exist, create it using a try-except block to handle the case where it already exists
+        if not schema_exists:
+            try:
+                cursor.execute(f"CREATE SCHEMA {map_state['import_schema']}")
+                print("Schema created successfully")
+            except psycopg2.Error as e:
+                print(f"Error creating schema: {e}")
+        else:
+            print("Schema already exists")
+
+        # commit the changes and close the cursor and connection
+        pg.commit()
+        cursor.close()
+        pg.close()
+
+    return map_state
+
+
+def clean_up_import_schema(map_state):
+    with SshKeyTempDir() as key_directory:
+        write_key_to_file(key_directory + "/id_ed25519", DB_BASTION_HOST_SSH_PRIVATE_KEY + "\n") 
+        ssh_tunnel = SSHTunnelForwarder(
+            (DB_BASTION_HOST),
+            ssh_username=DB_BASTION_HOST_SSH_USERNAME,
+            ssh_private_key=f"{key_directory}/id_ed25519",
+            remote_bind_address=(DB_RDS_HOST, 5432),
+        )
+        ssh_tunnel.start()
+
+        pg = psycopg2.connect(
+            host="localhost",
+            port=ssh_tunnel.local_bind_port,
+            user=DB_USER,
+            password=DB_PASS,
+            dbname=DB_NAME,
+            sslmode=DB_SSL_REQUIREMENT,
+            sslrootcert="/root/rds-combined-ca-bundle.pem",
+        )
+
+        cursor = pg.cursor()
+        sql = f"DROP SCHEMA IF EXISTS {map_state['import_schema']} CASCADE "
+        cursor.execute(sql)
+
+        pg.commit()
+        cursor.close()
+        pg.close()
+
+    return map_state
+
+# these temp directories are used to store ssh keys, because they will
+# automatically clean themselves up when they go out of scope.
+class SshKeyTempDir:
+    def __init__(self):
+        self.path = None
+
+    def __enter__(self):
+        self.path = tempfile.mkdtemp(dir='/tmp')
+        return self.path
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        shutil.rmtree(self.path)
+
+def write_key_to_file(path, content):
+    # Open the file with write permissions and create it if it doesn't exist
+    fd = os.open(path, os.O_WRONLY | os.O_CREAT, 0o600)
+
+    # Write the content to the file
+    os.write(fd, content.encode())
+
+    # Close the file
+    os.close(fd)
```

Comment:
Both of these functions are used to provide a place to briefly store SSH keys to disk because SSH insists on getting that way. This gives a place to store them and writes them out with the correct file permissions.